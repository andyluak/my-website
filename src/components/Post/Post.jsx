import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

import Tag from '../../components/Tag/Tag';

export default function Post({ index, post }) {
	let imageUrl = `https://portfoliostorage.fra1.digitaloceanspaces.com/${post.featuredImage}`;
	// Convert spaces in url to %20
	imageUrl = imageUrl.replace(/\s/g, '%20');

	return (
		<div className="post" key={index}>
			<div
				className="post-img"
				style={{
					backgroundImage: `url(${imageUrl})`,
				}}
			/>
			<h1>
				<Link to={`/${post.slug}`}>{post.title}</Link>
			</h1>

			<p
				className="post-content"
			>
				{post.content.text.substring(0, 40)}...
			</p>

			<div className="post-meta-container">
				<div className="post-author">
					<p className="author">{post.author}</p>
					<p className="date">
						{moment(post.createdAt).format('MMMM Do YYYY')}
					</p>
				</div>

				<div className="tags">
					{post.tags.map((tag, index) => {
						return <Tag key={index} tag={tag} />;
					})}
				</div>
			</div>
		</div>
	);
}
