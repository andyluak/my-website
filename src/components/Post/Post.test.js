import React from 'react';
import { shallow } from 'enzyme';
import Post from './Post';

describe( 'Post component tests', () => {
	it( 'excepct to render Post component', () => {
		const mockPost = {
			title: 'Test title',
			slug: 'test-slug',
			content: {
				rendered: '<p>Test content</p>',
				text: 'Test content'
			},
			featuredImage: 'code.jpg',
			date: '2019-01-01',
			tags: [ 'javascript'],

		}
		expect(shallow(<Post index={0} post={mockPost}/>)).toMatchSnapshot();
	})
})

