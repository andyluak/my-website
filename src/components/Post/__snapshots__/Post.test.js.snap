// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component tests excepct to render Post component 1`] = `
<div
  className="post"
  key="0"
>
  <div
    className="post-img"
    style={
      Object {
        "backgroundImage": "url(https://portfoliostorage.fra1.digitaloceanspaces.com/code.jpg)",
      }
    }
  />
  <h1>
    <Link
      to="/test-slug"
    >
      Test title
    </Link>
  </h1>
  <p
    className="post-content"
  >
    Test content
    ...
  </p>
  <div
    className="post-meta-container"
  >
    <div
      className="post-author"
    >
      <p
        className="author"
      />
      <p
        className="date"
      >
        February 26th 2022
      </p>
    </div>
    <div
      className="tags"
    >
      <Tag
        key="0"
        tag="javascript"
      />
    </div>
  </div>
</div>
`;
